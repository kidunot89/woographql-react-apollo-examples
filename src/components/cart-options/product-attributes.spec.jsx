import React from 'react';
import '@testing-library/react/dont-cleanup-after-each';
import { render, cleanup } from '@testing-library/react';

import { mockProduct } from '../../testing/mocks';
import ProductAttributes from './product-attributes';

describe('Product variation attributes', () => {
  afterAll(() => cleanup());
  let variation = mockProduct.variations.nodes[0];
  const setVariation = jest.fn((v) => variation = v);

  const { baseElement } = render(
    <ProductAttributes
      variations={mockProduct.variations.nodes}
      attributes={mockProduct.attributes}
      setVariation={setVariation}
    />,
  );

  it('should render successfully', () => {
    expect(baseElement).toMatchSnapshot();
  });
});