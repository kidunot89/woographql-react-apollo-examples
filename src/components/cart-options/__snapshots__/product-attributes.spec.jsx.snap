// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product variation attributes should render successfully 1`] = `
<body>
  <div>
    <div>
      <div>
        <div
          class="font-bold"
        />
        <div>
          <label>
            <input
              checked=""
              class="mr-3 text-center"
              name="Print"
              type="radio"
              value="ankara"
            />
            Ankara
          </label>
        </div>
        <div>
          <label>
            <input
              class="mr-3 text-center"
              name="Print"
              type="radio"
              value="candy-skull"
            />
            Candy skull
          </label>
        </div>
      </div>
    </div>
  </div>
</body>
`;
